### ISSUE TITLE
Unexpected parse error in expression

### ISSUE THREAD

**Issue by int-e** [rust-lang/rust#159670]
Consider this (artificial) code ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=f00cd5f61308667ce9f47c87f7adac94), seen on IRC):

```rust
{{} == ()}
```

It gets rejected with a parse error:

```
error: expected expression, found `==`
 --> src/main.rs:2:9
  |
2 | {{} == ()};
  |     ^^ expected expression

```

But I think it should parse as a block expression:

* [BlockExpression](https://doc.rust-lang.org/reference/expressions/block-expr.html#grammar-BlockExpression)
  * '{'
  * [Statements](https://doc.rust-lang.org/reference/expressions/block-expr.html#railroad-Statements)
    * [ExpressionWithoutBlock](https://doc.rust-lang.org/reference/expressions.html#grammar-ExpressionWithoutBlock)
      * [ExpressionWithoutBlockNoAttrs](https://doc.rust-lang.org/reference/expressions.html#grammar-ExpressionWithoutBlockNoAttrs)
        * [OperatorExpression](https://doc.rust-lang.org/reference/expressions/operator-expr.html#railroad-OperatorExpression)
          * [ComparisonExpression](https://doc.rust-lang.org/reference/expressions/operator-expr.html#grammar-ComparisonExpression)
            * [Expression](https://doc.rust-lang.org/reference/expressions.html#grammar-Expression)
              * [ExpressionWithBlock](https://doc.rust-lang.org/reference/expressions.html#grammar-ExpressionWithBlock)
                * [ExpressionWithBlockNoAttrs](https://doc.rust-lang.org/reference/expressions.html#grammar-ExpressionWithBlockNoAttrs)
                  * [BlockExpression](https://doc.rust-lang.org/reference/expressions/block-expr.html#grammar-BlockExpression)
                     * '{'
                     * '}'
            * '=='
            * [Expression](https://doc.rust-lang.org/reference/expressions.html#grammar-Expression)
               * [ExpressionWithoutBlock](https://doc.rust-lang.org/reference/expressions.html#grammar-ExpressionWithoutBlock)
                 * [ExpressionWithoutBlockNoAttrs](https://doc.rust-lang.org/reference/expressions.html#grammar-ExpressionWithoutBlockNoAttrs)
                    * [TupleExpression](https://doc.rust-lang.org/reference/expressions/tuple-expr.html#railroad-TupleExpression)
                      * '('
                      * ')'
  * '}'
          
Instead, `rustc` appears to decide that the inner `{}` is the first statement of a block expression and then gets stuck trying to parse the remainder as an expression.

### Meta

Tested Rust versions: 1.97.1, 1.99.0-nightly (2026-07-20 87e5904f5eb6398af6b2)

**Comment by int-e** [comment]
Oh this is close relative of #54482, with a different type of operator expression.
